tests/test_ingest.py

```diff
@@ -31,74 +31,73 @@
 
 """
 import ast
+import random
 import sys
 import unittest
 from unittest.mock import patch
 
-from gcp_mocks import mock_storage_client, mock_storage_blob, mock_firestore_client
+from gcp_mocks import mock_storage_client, mock_storage_blob
+
+from google.cloud import firestore
+import google.auth.credentials
 
 sys.path.append("../ingest")
 from ingest_pipeline import create_parser, validate_arguments, IngestPipeline
 
 
-def mock_load_expression_data(self, *args, **kwargs):
-    """Enables overwriting normal function with this placeholder.
-    Returning the arguments enables tests to verify that the code invokes
-    this method with expected argument values.
+def _make_credentials():
+    return unittest.mock.Mock(spec=google.auth.credentials.Credentials)
 
-    TODO:
-    Use Google's official Firestore mock,
-    https://github.com/googleapis/google-cloud-ruby/blob/master/google-cloud-firestore/EMULATOR.md#google-cloud-firestore-emulator
 
-    This will enable us to also verify (and thus cover) loading code *outputs*,
-    unlike here where we merely give a way to verify loading code *inputs*.
-    Doing so via integration tests will isolate us from implementation changes.
-    """
-    self.load_expression_data_args = args
-    self.load_expression_data_kwargs = kwargs
+def get_random_study_accession():
+    '''Randomly seeds to avoid collisions in Firestore emulator instance'''
+    study_number = random.randint(1, 1_000_001)
+    return f'SCP{study_number}'  # E.g. SCP1324
 
 
-def get_nth_gene_models(n, models, mock_dir):
-    """Return Nth actual and expected gene models, using actual and mock data
+def get_nth_gene_docs(n, docs, mock_dir):
+    """Return Nth actual and expected gene documents, using actual and mock data
     """
-    # TODO: Dense loads models as a `list`, Mtx loads models as a `dict_values`
-    # It seems both would ideally load using the same type.  Reconcile.
-    if isinstance(models, list):
-        # For Dense
-        actual_model = models[n].__dict__
-    else:
-        # For Mtx
-        actual_model = list(models)[n].__dict__
+
+    # Firestore does not return results in the same order every time,
+    # so sort documents by name to enable comparing Nth query result.
+    docs = sorted(docs, key=lambda doc: doc['name'])
+
+    actual_doc = docs[n]
 
     # Uncomment to print out new baseline data
     # Process to update baselines is manual: copy and paste it into new file
     # TODO: Automate when reasonable
-    print("actual_model")
-    print(actual_model)
+    print("actual_doc")
```

Comment:
You could use an F-string here. But up to you.

Reply:
[Done!](https://github.com/broadinstitute/scp-ingest-pipeline/pull/26/commits/a766165a9823cc67f287e8ea26d6ec255c8fe09e#diff-9398b55cff89e452e5b4049aceaac50cR71)